src/mdns/mdns_avahi.cpp

```diff
@@ -1044,17 +1044,17 @@ void PublisherAvahi::ServiceSubscription::HandleResolveResult(AvahiServiceResolv
     DiscoveredInstanceInfo instanceInfo;
     bool                   resolved = false;
 
+    VerifyOrExit(
+        aEvent == AVAHI_RESOLVER_FOUND,
+        otbrLogErr("failed to resolve service: %s", avahi_strerror(avahi_client_errno(mPublisherAvahi->mClient))));
+    VerifyOrExit(aHostName != nullptr, otbrLogErr("host name is null"));
```

Comment:
don't need to move this line up?

Reply:
The line 1052 will read data from `aAddress`, if `aEvent` indicates a failure, `aAddress` could be `nullptr`.

Reply:
done